pkg/dockerregistry/server/manifestschema1handler.go

```diff
@@ -53,56 +53,75 @@ type manifestSchema1Handler struct {
 var _ ManifestHandler = &manifestSchema1Handler{}
 
 func (h *manifestSchema1Handler) FillImageMetadata(ctx context.Context, image *imageapiv1.Image) error {
```

Comment:
As it's our internal type, I'd like to propose to split this function into 3 parts:

    Layers(ctx context.Context) (layers []imageapi.ImageLayer, annotations map[string]string, error)
    Signatures(ctx context.Context) ([][]byte, error)
    Metadata(ctx context.Context, layers []imageapi.ImageLayer) (imageapi.DockerImage, err error)

The output of these functions should depend only on the input manifest. And we do not need to parse manifest data in this functions, because we already have `h.manifest` (in this case we can remove DockerImageManifest in caller).

These functions can be reused in pkg/image/apis/image.ImageWithMetadata.

Reply:
Good idea, I'll rework.

Reply:
Reworked. I modified the declarations a bit. PTAL